Luke7787/ImmaculateInventors
# Feature/be crypt pt2

Summary:
BeCrypt now works in our database when adding users

Testing:
- pull code
- `npm install`
-Add the .env files. one in the root one in express-backend - see discord for what to put in each
___
- open a new terminal window
- Go into express-backend
- `npm run dev`
- open a new terminal window
- Go to the root folder
- `npm start`
___
**Add a user in boomerang:**

`POST`: http://localhost:8000/users

**JSON:**
{
"username": "TestUser",
"password": "ImmaculateInventors123!",
"email": "bklingen@calpoly.edu",
"country": "United States",
"state": "California",
"zipcode": "93407",
"city": "San Luis Obispo",
"firstName": "BJ",
"lastName": "Klingenberg"
}

___
After you add the user to the database you should see on the right panel that the password is now salted and encrypted. 


- Copy the password "**_ImmaculateInventors123!_**"  (don't forget the exclamation mark (!) at the end)

___
- Head over to http://localhost:3000/
- click sign in
- Input the credentials: "TestUser" and "ImmaculateInventors123!" for the username and password respectively.
- If there are no error messages then it should have worked. The username and password might just disappear if we haven't pulled actually signing in into main yet.
